api-list.go

```diff
@@ -423,6 +423,213 @@ func (c Client) ListObjects(ctx context.Context, bucketName, objectPrefix string
 	return objectStatCh
 }
 
+// ListObjectVersions - List versions of some or all objects.
+//
+// e.g.:
+//   api := client.New(....)
+//   // Create a done channel.
+//   doneCh := make(chan struct{})
+//   defer close(doneCh)
+//   // Recurively list all objects in 'mytestbucket'
+//   recursive := true
+//   for message := range api.ListObjectVersions(ctx, "mytestbucket", "prefix", recursive, doneCh) {
+//       fmt.Println(message)
+//   }
+func (c Client) ListObjectVersions(ctx context.Context, bucketName, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectVersionInfo {
```

Comment:
We don't need another ListObjectVersions() API instead we only need one API 

```
ListObjects(ctx context.Context, bucketName, opts ListObjectsOptions)

type ListObjectsOptions {
    Type Version1/Version2
    WithVersions bool
    WithMetadata bool
    Prefix string
    Recursive bool
    Marker string
    MaxKeys int // represents max-keys to be fetched per request
```

We don't need doneCh anymore we use context everywhere that is the canceller not doneCh

Reply:
> We don't need another ListObjectVersions() API instead we only need one API
> 
> ```
> ListObjects(ctx context.Context, bucketName, opts ListObjectsOptions)
> 
> type ListObjectsOptions {
>     Type Version1/Version2
>     WithVersions bool
>     WithMetadata bool
>     Prefix string
>     Recursive bool
>     Marker string
>     MaxKeys int // represents max-keys to be fetched per request
> ```
> 
> We don't need doneCh anymore we use context everywhere that is the canceller not doneCh

Done